cnf-tests/pod-utils/cyclictest-runner/main.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"flag"
+	"k8s.io/klog"
+	"os/exec"
+	"strconv"
+	"time"
+
+	podutils "github.com/openshift-kni/cnf-features-deploy/cnf-tests/pod-utils"
+)
+
+const cyclictestBinary = "/usr/bin/cyclictest"
+
+func main() {
+	klog.InitFlags(nil)
+
+	var rtPriority = flag.String("rt-priority", "1", "specify the SCHED_FIFO priority (1-99)")
+	var duration = flag.String("duration", "15s", "specify a length for the test run. Append 'm', 'h', or 'd' to specify minutes, hours or days.")
+	var threads = flag.String("threads", "", "number of threads: without NUM, threads = max_cpus. without -t default = 1")
```

Comment:
same, if this is a number use flag.Int

Reply:
This flag can also be an empty string ```""```

Reply:
Can't you use 0 as default?

Reply:
Empty means all threads, as you can see in the description

Reply:
I can see the description, but it feels totally odd.
If you want a default, you can set it to -1 and still achieve the result you want, or even better fetch maxCpus before setting the flags and use that as default value, I think that might work

Reply:
> I can see the description, but it feels totally odd.
> If you want a default, you can set it to -1 and still achieve the result you want, or even better fetch maxCpus before setting the flags and use that as default value, I think that might work

IMHO

default should be 1 thread

to designate running on all cpus,
can either use 0 or the word MAX

the empty notion feels too weird

Reply:
I just mimic the behavior of the tool itself (also the description) but yes, I can totally make it more intuitive

Reply:
> I just mimic the behavior of the tool itself (also the description) but yes, I can totally make it more intuitive

That's a good conflict though.
whether we want our tool to be more intuitive or more closely mimic the underlying tool ?

How close are the rest of the parameters?

Reply:
The parameters descriptions are almost identical and also describe the default values the tool has.
There were other parameters that I added to the runner as "best-practices" that I saw being used in other projects, for example ```--histogram```

Reply:
> The parameters descriptions are almost identical and also describe the default values the tool has.
> There were other parameters that I added to the runner as "best-practices" that I saw being used in other projects, for example `--histogram`

so in that case, I think I vote for preserving the original flags and meaning. our description can be less cryptic if needed.
and maybe work upstream on changing them ?

Reply:
yes, if we believe `-t 0` is more intuitive and useful than `-t` we should talk with tool maintainers and see what they think if/when they fix it, we can do the same

Reply:
Just to clarify, for now, I change nothing in order to be aligned with the tool behavior

Reply:
We are mixing two things. The type of threads, and the behaviour we want.
If it's a number, the type should be int. 

If we want to have a specific behaviour for when the parameter is not passed, then we can have a default as 0 or -1 and react according to that (the same way we do with `""`)